app/episode/[ep_number]/page.tsx

```diff
@@ -20,60 +22,107 @@ export default function EpisodePage({
       const episodeNumber = parseInt(params.ep_number, 10);
       const res = await fetch(`/api/episode?episode_number=${episodeNumber}`);
       const json = await res.json();
-      setData(json["data"]);
-      console.log(json["data"]);
+      const sortedData = {
+        ...json["data"],
+        episode_data: json["data"].episode_data.sort(
+          (a: SegmentProps, b: SegmentProps) => {
+            const lengthA = a.end_time_ms - a.start_time_ms;
+            const lengthB = b.end_time_ms - b.start_time_ms;
+            return lengthB - lengthA;
```

Comment:
This looks very similar to the top3 in page select. I guess this is more general. Maybe I should take the difference for you.